docs/en/creating-widgets/supplemental.md

```diff
@@ -0,0 +1,891 @@
+```
+
+By default, running the command for a given `dojo create widget --name MyScaffoldedWidget` will create the following structure:
+
+    myscaffoldedwidget/
+    ├── MyScaffoldedWidget.ts
+    ├── styles/
+    │   ├── myscaffoldedwidget.m.css
+    │   └── myscaffoldedwidget.m.css.d.ts
+    └── tests/
+        └── unit
+            └── MyScaffoldedWidget.ts
+
+The location where styles and tests are created can be customized using the `--styles` and `--tests` arguments respectively.
+
+A [Custom Element](https://www.w3.org/TR/2016/WD-custom-elements-20161013/) descriptor will be generated if the `--component` argument is passed. When doing so, the scaffolded widget will also include an appropriate `@customElement` decorator with an empty template object, ready for further configuration.
+
+# Rendering in Dojo
+
+Dojo is a reactive framework, handling responsibilities of data change propagation and associated rendering updates behind the scenes. It uses a virtual DOM (VDOM) concept to represent elements intended for output, with nodes in the VDOM being simple JavaScript objects that are designed to be more efficient to work with than actual DOM elements.
+
+Applications only need to concern themselves with declaring their intended output structure as a hierarchy of virtual DOM nodes, typically done as the return values from their [widgets' render functions](#basic-widget-structure). The framework's [`Renderer`](#rendering-to-the-dom) component then synchronizes the intended output with concrete elements in the DOM. Virtual DOM nodes also serve to configure and provide state to widgets and elements, by passing in properties.
+
+Dojo supports subtree rendering, meaning that when a change in state occurs, the framework is able to determine specific subsets of VDOM nodes affected by the change. Only the required corresponding subtrees within are then updated to reflect the change, increasing rendering performance and resulting in an improved user interactive experience.
+
+> **Note:** Returning virtual nodes from widget render functions is the only concern applications have around rendering. Attempting to use any other practice [is considered an anti-pattern in Dojo application development](#widget-development-best-practices), and should be avoided.
+
+## TSX Support
+
+Dojo supports use of the `jsx` syntax extension known as [`tsx` in TypeScript](https://www.TypeScriptlang.org/docs/handbook/jsx.html). This syntax allows for a more convenient representation of a widget's VDOM output that is closer to the resulting HTML within a built application.
+
+### TSX-enabled applications
+
+TSX-enabled projects can easily be scaffolded via the [`dojo create app --tsx` CLI command](https://github.com/dojo/cli-create-app).
+
+For Dojo projects that were not scaffolded in this way, TSX can be enabled with the following additions to the project's TypeScript config:
+
+> `./tsconfig.json`
+
+```json
+{
+	"compilerOptions": {
+		"jsx": "react",
```

Comment:
Note, we should file an enhancement request on the TS team to add a dojo option here. :)